mesonbuild/interpreter/type_checking.py

```diff
@@ -577,6 +597,17 @@ def _name_validator(arg: T.Optional[T.Union[str, T.List]]) -> T.Optional[str]:
         validator=in_set_validator(set(compilers.all_languages)),
         since='0.51.0',
     ),
+    KwargInfo(
+        'objects',
+        ContainerTypeInfo(list, (str, File, CustomTarget, CustomTargetIndex, GeneratedList)),
```

Comment:
We permit ExtractedObjects as well, but don't make it possible to use in KwargInfo types?